lib/DebugApi.js

```diff
@@ -201,4 +206,140 @@ DebugApi.prototype.run = function() {
     }).listen(this.port);
 }
 
+DebugApi.prototype.killPortal = Promise.coroutine(function*(req, response) {
+    const result = {error: [], stages: []};
+    const body = yield this._wrapJsonReq(req, response);
+    // Room room_id to lookup and delete the alias from.
+    const roomId = body["room_id"];
+    // IRC server domain
+    const domain = body["domain"];
+    // IRC channel
+    const channel = body["channel"];
+    // Should we tell the room about the deletion. Defaults to true.
+    const notice = !(body["leave_notice"] === false);
+    // Should we remove the alias from the room. Defaults to true.
+    const remove_alias = !(body["remove_alias"] === false);
+
+    ["room_id", "channel", "domain"].forEach((key) => {
+        if (typeof(body[key]) !== "string") {
+            result.error.push(`'${key}' is missing from body or not a string`);
+        }
+    });
+    if (result.error.length > 0) {
+        this._wrapJsonResponse(result.error, false, response);
+        return;
+    }
+
+    log.info(
+`Requested deletion of portal room alias ${roomId} through debug API
+Domain: ${domain}
+Channel: ${channel}
+Leave Notice: ${notice}
+Remove Alias: ${remove_alias}`);
+
+    // Find room
+    let room = yield this.ircBridge.getStore().getRoom(
+        roomId,
+        domain,
+        channel,
+        "alias"
+    );
+    if (room === null) {
```

Comment:
Would this also catch room being `undefined`?

Reply:
The function only returns Room or null. :)